### PR TITLE
4078: Standardize 'comments' and 'description' Fields

### LINKED ISSUE #4078: Standardize the use of name, description, and comments fields
### Proposed Changes
We need to do an audit of all existing `description` and `comments` fields and adjust where necessary to ensure consistency across all models. I'm not sure this has ever been formally defined, but in my mind the distinction is:

* `description`: A one-liner of limited length describing the object, suitable for inclusion when viewing a list of objects
* `comments`: Long-form arbitrary remarks regarding an object, complete with Markdown support

### Justification
This ensures consistency in the way the fields are handled across all models.

### PR DESCRIPTION
**Pull request by dstarner**
### Fixes: #4078 

**NOTE**: This is merging into the `develop-2.8` branch, because this is a 2.8 milestone issue.

Implements a `DescriptionModel` and `CommentModel` that standardizes the `comments` and `description` fields for any/all models. They are broken out into model extensions / mixins to make it easier to standardize across the application(s) for current *and* future models.

The migration files were added as `<id>_standarize_description_comment_fields.py` which I felt was an appropriate name for them.

---
I initially called them `...Mixins` but when being appended to some of the more basic models, this caused a [method resolution order exception](https://stackoverflow.com/questions/3003053/metaclass-multiple-inheritance-inconsistency) so I called them `Model`s because `MyModel(models.Model, CommentModel)` had to become `MyModel(CommentModel)`.

### CODE REVIEW

**Comment by dstarner** [comment]
I can add a comment to the changelog before this gets reviewed / lands. I left that out

**Comment by dstarner** [comment]
I also need to fix this branch to go into `develop-2.8`

File: netbox/utilities/models.py
Comment (jeremystretch) [inline netbox/utilities/models.py]: Using abstract models to add individual fields is overkill. At most we would implement a custom field class, and even that seems excessive. All we need to do is standardize and document the field attributes.

**Review by jeremystretch — CHANGES_REQUESTED** [review]

File: netbox/utilities/models.py
Comment (DanSheps) [reply netbox/utilities/models.py]: Perhaps just go with a simple constant for length, that looks to be all that we need to standardize on.

**Comment by dstarner** [comment]
Fair, overengineered, will fix

**Comment by jeremystretch** [comment]
@dstarner Hold off on this for now, please. There may be some additional work to do: see my comment on #4078.

**Pull request closed without merging**
